pkg/controller/cronjob/cronjob_controller2.go

```diff
@@ -102,9 +103,38 @@ func NewController2(jobInformer batchv1informers.JobInformer, cronJobsInformer b
 		AddFunc: func(obj interface{}) {
 			jm.enqueueController(obj)
 		},
-		UpdateFunc: func(_, curr interface{}) {
+		UpdateFunc: func(old, curr interface{}) {
 			// TODO: @alpatel need to figure out if inspection and queueing of old cronjob
 			//  is required.
+			//  the big cases that needs to be handled
+			//  1. change in CJ.spec.schedule
+			//  2. change in CJ.spec.JobTemplate
+			//  other CJ.spec params seem to be dealt with by the controller in sync loop on updates
+			oldCJ, okOld := old.(*batchv1beta1.CronJob)
+			newCJ, okNew := old.(*batchv1beta1.CronJob)
+
+			if !okOld || !okNew {
+				// typecasting of one failed, handle this better, may be log entry
+				return
+			}
+			if oldCJ.Spec.Schedule != newCJ.Spec.Schedule {
+				// schedule changed, change the requeue time
+				sched, err := cron.ParseStandard(newCJ.Spec.Schedule)
+				if err != nil {
+					// this is likely a user error in defining the spec value
+					// we should log the error and not reconcile this cronjob until an update to spec
+					klog.Errorf("unparseable schedule: %s : %s", newCJ.Spec.Schedule, err)
+					return
+				}
+				now := time.Now()
+				t := time.Duration(int64(sched.Next(now).Sub(now).Seconds()))
+				jm.enqueueControllerWithTime(curr, t)
```

Comment:
@soltysh the way I am trying to think about the problem here is that, if the change is schedule results in next requeue having to be sooner than it already was, it will be handled here by the queue. If the next requeue is further than previous schedule, the sync loop will essentially be a no-op for the already queued key with old schedule.